tests/test_defects.py

```diff
@@ -0,0 +1,270 @@
+
+        # now test the AND and OR modes
+        defectList = [defects]*10  # 10 identical defect sets
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'AND')
+        self.assertEqual(defects, merged)
+
+        defectList[7] = defectList[7][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'AND')
+        self.assertNotEqual(defects, merged)
+
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'OR')
+        self.assertEqual(defects, merged)
+
+    def test_pixelCounting(self):
+        config = copy.copy(self.defaultConfig)
+        config.nPixBorderUpDown = 0
+        config.nPixBorderLeftRight = 0
+        task = cpPipe.defects.FindDefectsTask(config=config)
+        defects = task.findHotAndColdPixels(self.flatExp, 'flat')
+
+        defectArea = 0
+        for defect in defects:
+            defectArea += defect.getBBox().getArea()
+
+        crossCheck = 0
+        for x, y, s in self.brightDefects:
+            crossCheck += s**2
+        for x, y, s in self.darkDefects:
+            crossCheck += s**2
+
+        self.assertEqual(defectArea, task._nPixFromDefects(defects))  # the real function being tested
+        self.assertEqual(defectArea, crossCheck)  # but calculate the sum two ways
+
+    def test_getNumGoodPixels(self):
+        testImage = self.flatExp.clone()
+        mi = testImage.maskedImage
+
+        imageSize = testImage.getBBox().getArea()
+        nGood = self.defaultTask._getNumGoodPixels(mi)
+
+        self.assertEqual(imageSize, nGood)
+
+        NODATABIT = mi.mask.getPlaneBitMask("NO_DATA")
+
+        noDataBox = Box2I(Point2I(31, 49), Extent2I(3, 6))
+        testImage.mask[noDataBox] |= NODATABIT
+
+        self.assertEqual(imageSize - noDataBox.getArea(), self.defaultTask._getNumGoodPixels(mi))
+        self.assertEqual(imageSize, self.defaultTask._getNumGoodPixels(mi, 'BAD'))  # check for misfire
+
+        testImage.mask[noDataBox] ^= NODATABIT  # XOR to reset what we did
+        self.assertEqual(imageSize, nGood)
+
+        BADBIT = mi.mask.getPlaneBitMask("BAD")
+        badBox = Box2I(Point2I(85, 98), Extent2I(4, 7))
+        testImage.mask[badBox] |= BADBIT
+
+        self.assertEqual(imageSize - badBox.getArea(), self.defaultTask._getNumGoodPixels(mi, 'BAD'))
+
+    def test_edgeMasking(self):
+        testImage = self.flatExp.clone()
+        mi = testImage.maskedImage
+
+        self.assertEqual(countMaskedPixels(mi, 'EDGE'), 0)
+        self.defaultTask._setEdgeBits(mi)
+
+        hEdge = self.defaultConfig.nPixBorderLeftRight
+        vEdge = self.defaultConfig.nPixBorderUpDown
+        xSize, ySize = mi.getDimensions()
+
+        nEdge = xSize*vEdge*2 + ySize*hEdge*2 - hEdge*vEdge*4
+
+        self.assertEqual(countMaskedPixels(mi, 'EDGE'), nEdge)
+
+    def testImport(self):
+        import lsst.cp.pipe as cpPipe  # noqa: F401
+
+    def testClassInstantiation(self):
```

Comment:
Also superfluous: you instantiate these in `setUp`, and tests that something doesn't blow up on instantiation aren't generally useful without also testing some functionality.

Reply:
Yeah, this was written before the substantive tests. Removed.